### PR TITLE
fix(sql): fix invalid column error in group by with join

### LINKED ISSUE #4534: Alias in GROUP BY clause is not recognized
### To reproduce

This issue is reported by a person who's using QuestDB [Cube.js](https://cube.dev/) driver.

```sql
CREATE TABLE 'dim_apTemperature' (id INT, category STRING, aparent_temperature FLOAT);

CREATE TABLE 'fact_table' (
  id INT,
  date TIMESTAMP,
  fulltime STRING,
  id_OF INT,
  id_companie INT,
  id_location INT,
  id_asset INT,
  energy_power FLOAT,
  energy_power_factor FLOAT,
  data_time TIMESTAMP,
  id_state INT,
  temperature FLOAT,
  radiation FLOAT,
  id_wind_direction INT,
  pressure FLOAT,
  precipitation FLOAT,
  id_precipitation INT,
  wind_intensity FLOAT,
  id_wind_intensity INT,
  humidity FLOAT,
  id_humidity INT,
  dew_point FLOAT,
  cloud_base_height FLOAT,
  id_cloud INT,
  thermal_sensation FLOAT,
  id_aparent_temperature INT,
  ST FLOAT
);
```

Execute the following query:
```sql
SELECT
  "dim_ap_temperature".category "dim_ap_temperature__category",
  timestamp_floor('d', to_timezone("fact_table".data_time, 'UTC')) "fact_table__date_time_day",
  avg("fact_table".radiation) "fact_table__avg_radiation",
  avg("fact_table".energy_power) "fact_table__energy_power"
FROM
  fact_table AS "fact_table"
  LEFT JOIN dim_apTemperature AS "dim_ap_temperature" ON "fact_table".id_aparent_temperature = "dim_ap_temperature".id
GROUP BY
  "dim_ap_temperature__category",
  "fact_table__date_time_day"
ORDER BY
  "fact_table__avg_radiation" DESC
LIMIT
  10000;
```
and get this error:
```
Invalid column: fact_table.data_time
```

If you remove GROUP BY clause completely, the query executes fine.

### QuestDB version:

latest master

### OS, in case of Docker specify Docker and the Host OS:

Linux

### File System, in case of Docker specify Host File System:

ext4

### Full Name:

Andrei Pechkurov

### Affiliation:

QuestDB

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [X] Yes, I have

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Fixes #4534

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 44 / 44 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|44|44|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 44 / 44 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|44|44|100.00%|

**Review by bluestreak01 — CHANGES_REQUESTED** [review]
we need to find a better way to do it, rather than step 2 undoing work of step 1

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 44 / 44 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|44|44|100.00%|

**Comment by puzpuzpuz** [comment]
> we need to find a better way to do it, rather than step 2 undoing work of step 1

I've updated the patch so that we don't redo anything. PTAL

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|20|20|100.00%|

File: core/src/test/java/io/questdb/test/griffin/GroupByTest.java [hunk 1]
```diff
@@ -1404,6 +1404,124 @@ public void testGroupByWithDuplicateSelectColumn() throws Exception {
...
+                    "create table fact_table as (" +
+                            "  select x::int id_aparent_temperature," +
+                            "         (x * 120000000)::timestamp date_time," +
+                            "         rnd_float() radiation," +
+                            "         rnd_float() energy_power" +
+                            "  from long_sequence(10)" +
+                            ");"
+            );
+
+            final String expectedResult = "dim_ap_temperature__category\tfact_table__date_time_day\tfact_table__avg_radiation\tfact_table__energy_power\n" +
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/test/griffin/GroupByTest.java:1428]: nit: we should use "assertSql" (it is simpler and faster than "assertQuery") and also assert execution plan.

Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/test/griffin/GroupByTest.java:1428]: Done in bfa49e7c6fcdc9ba8a448a0726003cb51560e0ed

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|20|20|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|20|20|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**